src/mpyl/build.py

```diff
@@ -95,24 +47,26 @@ def run_mpyl(run_properties: RunProperties) -> RunResult:
     logger = logging.getLogger("mpyl")
     start_time = time.time()
     try:
-        run_result = RunResult(run_properties=run_properties)
+        run_result = RunResult(run_properties=run_properties, run_plan=run_plan)
 
         if not run_result.has_projects_to_run(include_cached_projects=False):
             logger.info("Nothing to do. Exiting..")
             return run_result
 
         logger.info("Run plan:")
-        console.print(Markdown(f"\n\n{run_result_to_markdown(run_result)}"))
+        console.print(Markdown(f"\n\n{run_result.status_line}  \n"))
+        run_plan.print_markdown(console, run_properties.stages)
 
         try:
-            steps = Steps(
+            executor = Steps(
```

Comment:
I had the exact same internal struggle, I left this to trigger this discussion 😂 

The existing naming scheme in MPyL is a bit confusing:
* an instance of the class `Step` (e.g. `Sbt Build` or `Kubernetes Deploy`) is called called "step" and "executor" (in code and documentation)
* a collection of `Step`s is called `steps` (in code) 
* the only instance of the class `Steps` is called `steps` in code, but `executor` in the documentation 🤯

I fully agree with you: personally I dislike the name `Steps` and I'd much rather align code and documentation and call it `Executor` (since it executes one more `Step`s). I didn't want to rename the entire class without some context.

I'll apply that change 👍